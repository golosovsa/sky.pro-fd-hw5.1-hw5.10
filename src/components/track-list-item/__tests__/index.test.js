import ShallowRenderer from 'react-test-renderer/shallow'
import TrackListItem from '..'

describe('<TrackListItem />', () => {
  describe('Snapshot test', () => {
    it('should render successfully and match snapshot', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(
        <TrackListItem item={{ name: 'name', author: 'author', album: 'album' }} />
      )

      expect(tree).toMatchSnapshot()
    })

    it('should render successfully and match snapshot is selected', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(
        <TrackListItem selected={true} item={{ name: 'name', author: 'author', album: 'album' }} />
      )

      expect(tree).toMatchSnapshot()
    })

    it('should render successfully and match snapshot is liked', () => {
      const renderer = new ShallowRenderer()
      const tree = renderer.render(
        <TrackListItem liked={true} item={{ name: 'name', author: 'author', album: 'album' }} />
      )

      expect(tree).toMatchSnapshot()
    })
  })
})
