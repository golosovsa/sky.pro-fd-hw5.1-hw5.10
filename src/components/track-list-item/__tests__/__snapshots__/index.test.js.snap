// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackListItem /> Snapshot test should render successfully and match snapshot 1`] = `
<styled.div
  onClick={[Function]}
>
  <TrackListItemIcon />
  <TrackListItemName
    name="name"
  />
  <styled.p>
    author
  </styled.p>
  <styled.p>
    album
  </styled.p>
  <Styled(Component)
    onClick={[Function]}
  />
  <styled.p>
    NaN:NaN
  </styled.p>
</styled.div>
`;

exports[`<TrackListItem /> Snapshot test should render successfully and match snapshot is liked 1`] = `
<styled.div
  onClick={[Function]}
>
  <TrackListItemIcon />
  <TrackListItemName
    name="name"
  />
  <styled.p>
    author
  </styled.p>
  <styled.p>
    album
  </styled.p>
  <Styled(Component)
    active={true}
    onClick={[Function]}
  />
  <styled.p>
    NaN:NaN
  </styled.p>
</styled.div>
`;

exports[`<TrackListItem /> Snapshot test should render successfully and match snapshot is selected 1`] = `
<styled.div
  onClick={[Function]}
  selected={true}
>
  <TrackListItemIcon />
  <TrackListItemName
    name="name"
  />
  <styled.p>
    author
  </styled.p>
  <styled.p>
    album
  </styled.p>
  <Styled(Component)
    onClick={[Function]}
  />
  <styled.p>
    NaN:NaN
  </styled.p>
</styled.div>
`;
